tests/clpy_tests/opencl_tests/ultima_tests/test_half.py

```diff
@@ -0,0 +1,63 @@
+# flake8: noqa
+# TODO(vorj): When we will meet flake8 3.7.0+,
+#               we should ignore only W291 for whole file
+#               using --per-file-ignores .
+
+import unittest
+import utility
+
+
+class TestUltimaTrickingHalf(unittest.TestCase):
+
+    def test_type_half(self):
+        x = '''
+__clpy__half f() 
+{
+    __clpy__half a;
+    return (__clpy__half)(42.F);
+}
+'''
+        y = utility.exec_ultima(
+            '''
+            half f(){
+              half a;
+              return static_cast<half>(42.f);
+            }
+            ''')
+        self.assertEqual(x[1:], y)
+
+    def test_variable_named_half(self):
+        x = '''
+void f() 
+{
+    int __clpy__half = 1 / 2;
```

Comment:
Is this really the expected result?  `__clpy__harl` is name of type, therefore I think ultima must not replace this variable name.

Reply:
**YES**, this is the expected result.
In OpenCL C, `half` is reserved, thus we cannot use this identifier as a variable name.
However, as it is possible to use it in CUDA C++, [Chainer contains the variable named `half`](https://github.com/chainer/chainer/blob/3af946d4be96d256cff274a35cbcc039fc70aab0/chainer/functions/activation/swish.py#L25).
This is the reason why I implemented this `half`-trick in Ultima.

Reply:
OK